### ISSUE TITLE
unnecessary sign change?

### ISSUE THREAD

**Issue by bradybus** [vvhitedog/three_sphere_intersection#1]
https://github.com/vvhitedog/three_sphere_intersection/blob/79702de1229a717376716d384d2ee7048500508b/three_sphere_intersection.py#L155

assigning `x = -x` and using it with `+x` in the equation for `c` seems the same as not changing the sign and using it with `-x` 2 lines farther down. Am I missing something?

**Comment by vvhitedog** [comment]
Yes, I think so. This was a while ago, but I think that when d<r0 then one sphere fits inside the other. I had thought this required two distinct cases (when sphere fits and when sphere doesn't), but it's possible that both are handled by the same set of equations.

**Comment by bradybus** [comment]
[circle_center.pdf](https://github.com/vvhitedog/three_sphere_intersection/files/5859869/circle_center.pdf)

I did some more checking, working out the equations on paper, and checking a few situations. See the graphic.
The code currently checks `if d<r0: x=-x`, but the graphic shows that with `d<ro`, the x should sometimes be added, and sometimes subtracted. Besides, the sign gets changed twice, which ends with the same result either way.
I think the `if` statement should be removed and just use the value `x` without altering it, and use `c=c0 + (n/d) * (d-x)`.

And you beat me to it. :) I'll submit a pull request in a bit, maybe tomorrow morning, with some enhancements, and an upgrade to py3, if you don't mind.

**Comment by vvhitedog** [comment]
Sure, any improvement/fix is appreciated, thanks :)
